_posts/2019-12-17-why-graalvm-19-2.adoc

```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: 'Why Quarkus 1.1.0.Final will still use GraalVM 19.2'
+date: 2019-12-17
+tags: quarkus
+synopsis: For Quarkus 1.1.0.Final, we had to take the hard decision to go back to GraalVM 19.2 while our CR1 used 19.3. Here is why. 
+author: gsmet
+---
+
+One of the highly anticipated features of Quarkus 1.1.0.CR1 was the GraalVM 19.3 support. GraalVM 19.3 changes quite a lot of things so it was a all or nothing update: it wasn't possible to support GraalVM 19.2 and 19.3 together.
+
+Supporting 19.3 was quite a struggle. We had several regressions, some due to Quarkus not doing the right thing (we fixed it), some due to GraalVM regressions (we worked hand in hand with the GraalVM team to fix them). Considering the next micro bugfix release of GraalVM is planned for mid-January, we were on our own to work around the GraalVM issues by tweaking Quarkus and writing substitutions.
+
+Even if it wasn't pretty, the end result was solid enough that we decided to ship CR1 with GraalVM 19.3 support.
+
+And, yet, our 1.1.0.Final is back to GraalVM 19.2. It was a hard decision, believe us, especially with all the work invested to get things ready for 1.1. So why this step back?
+
+== We don't want to break our promise
+
+One of Quarkus promises is that your Quarkus application running with the JVM could be compiled to a native executable with no hassle.
+
+Due to a number of issues, it wasn't the case anymore:
+
+ * We have https://github.com/oracle/graal/issues/1927[a concurrency issue striking randomly] meaning sometimes your native image build will fail with no good reason;
+ * Some Java code triggers a number of low-level issues: https://github.com/oracle/graal/issues/1995 broke our Neo4j extension (and more generally Neo4j support in GraalVM), with https://github.com/oracle/graal/issues/1971, we were not able to build native images for some of our Camel Quarkus integration tests. While we could have lived with the latter, it also means it could happen to any of your application.
```

Comment:
```suggestion
 * Some Java code triggers a number of low-level issues: https://github.com/oracle/graal/issues/1995 broke our Neo4j extension (and more generally Neo4j support in GraalVM), with https://github.com/oracle/graal/issues/1971, we were not able to build native images for some of our Camel Quarkus integration tests. While we could have lived with the latter, it also means it could happen to any of your applications.
```